enterprise/models.py

```diff
@@ -541,3 +541,81 @@ def __repr__(self):
         Return uniquely identifying string representation.
         """
         return self.__str__()
+
+
+@python_2_unicode_compatible
+class EnterpriseCourseEnrollment(TimeStampedModel):
+    """
+    Store information about the enrollment of a user in a course.
+
+    This model is the central source of truth for information about
+    whether a particular user, linked to a particular EnterpriseCustomer,
+    has been enrolled in a course, and is the repository for any other
+    relevant metadata about such an enrollment.
+    """
+
+    class Meta(object):
+        unique_together = (('user', 'course_id',),)
+        app_label = 'enterprise'
+
+    user = models.ForeignKey(
+        EnterpriseCustomerUser,
+        blank=False,
+        null=False,
+        related_name='enterprise_enrollments',
+        help_text=_(
+            "The enterprise learner to which this enrollment is attached."
+        )
+    )
+    consent_granted = models.NullBooleanField(
+        help_text=_(
+            "Whether the learner has granted consent for this particular course."
+        )
+    )
+    course_id = models.CharField(
```

Comment:
In original discussion that we had with @brittneyexline we talked about this model essentially holding the link between an EnterpriseCustomerUser and a [CourseEnrollment](https://github.com/edx/edx-platform/blob/master/common/djangoapps/student/models.py#L969) via their identifiers.  Just wondering if you considered doing that first and opted instead for the straight `course_id` and what led you to decide that was the better route?